SmartDeviceLink_Example/SubscribeButtonManager.swift

```diff
@@ -27,50 +26,28 @@ class SubscribeButtonManager {
         }
 
         presetButtons.forEach { buttonName in
-            guard presetButtonSubscriptionIDs[buttonName] == nil else {
-                SDLLog.w("The app is already subscribed to the \(buttonName.rawValue.rawValue) button")
-                return
-            }
-
-            let subscriptionID = sdlManager.screenManager.subscribeButton(buttonName) { [weak self] (press, event, error) in
-                guard error == nil, let press = press else {
-                    SDLLog.e("Error subscribing to the \(buttonName.rawValue.rawValue) button")
-                    self?.presetButtonSubscriptionIDs[buttonName] = nil;
-                    return
-                }
-
-                let alert: SDLAlert
-                if press.buttonPressMode == .short {
-                    alert = AlertManager.alertWithMessageAndCloseButton("\(press.buttonName.rawValue.rawValue) pressed")
-                } else {
-                    alert = AlertManager.alertWithMessageAndCloseButton("\(press.buttonName.rawValue.rawValue) long pressed")
-                }
-
-                self?.sdlManager.send(alert)
-            }
-
-            presetButtonSubscriptionIDs[buttonName] = subscriptionID
+            sdlManager.screenManager.subscribeButton(buttonName, withObserver: self, selector: #selector(buttonPressEvent(buttonName:error:buttonPress:buttonEvent:)))
         }
     }
 
-    /// Unsubscribes to all subscribed preset buttons.
-    func unsubscribeToPresetButtons() {
-        guard !presetButtonSubscriptionIDs.isEmpty else {
-            SDLLog.w("The app is not subscribed to any preset buttons")
+    @objc private func buttonPressEvent(buttonName: SDLButtonName, error: Error?, buttonPress: SDLOnButtonPress?, buttonEvent: SDLOnButtonEvent?) {
+        guard error == nil else {
+            SDLLog.e("There was an error subscribing to the preset button: \(error!.localizedDescription)")
             return
         }
 
-        for (buttonName, subscriptionID) in presetButtonSubscriptionIDs {
-            guard let subscriptionID  = subscriptionID as? NSObject else { continue }
-            sdlManager.screenManager.unsubscribeButton(buttonName, withObserver: subscriptionID) { [weak self] (error) in
-                guard error == nil else {
-                    SDLLog.e("The \(buttonName.rawValue.rawValue) button was not unsubscribed successfully")
-                    return
-                }
+        guard let buttonPress = buttonPress else { return }
```

Comment:
Again, if we're doing this, why get the button event?

Reply:
I removed the button event param from the selector